File: saim-backend/app/schemas/zona.py
from pydantic import BaseModel, ConfigDict
from typing import Optional
from datetime import datetime

class ZonaBase(BaseModel):
    codigo: str
    nombre: str
    descripcion: Optional[str] = None
    activo: bool = True

class ZonaCreate(ZonaBase):
    creado_por: int
    actualizado_por: int

class ZonaUpdate(BaseModel):
    codigo: Optional[str] = None
    nombre: Optional[str] = None
    descripcion: Optional[str] = None
    activo: Optional[bool] = None
    actualizado_por: int

class ZonaResponse(ZonaBase):
    id_zona: int
    creado_en: datetime
    creado_por: int
    actualizado_en: datetime
    actualizado_por: int

    model_config = ConfigDict(from_attributes=True)

File: saim-frontend/lib/features/catalogs/models/zona.dart
class Zona {
  final int? idZona;
  final String codigo;
  final String nombre;
  final String? descripcion;
  final bool activo;
  final DateTime? creadoEn;
  final int? creadoPor;
  final DateTime? actualizadoEn;
  final int? actualizadoPor;

  Zona({
    this.idZona,
    required this.codigo,
    required this.nombre,
    this.descripcion,
    this.activo = true,
    this.creadoEn,
    this.creadoPor,
    this.actualizadoEn,
    this.actualizadoPor,
  });

  factory Zona.fromJson(Map<String, dynamic> json) {
    return Zona(
      idZona: json['id_zona'],
      codigo: json['codigo'] ?? '',
      nombre: json['nombre'] ?? '',
      descripcion: json['descripcion'],
      activo: json['activo'] ?? true,
      creadoEn: json['creado_en'] != null ? DateTime.parse(json['creado_en']) : null,
      creadoPor: json['creado_por'],
      actualizadoEn: json['actualizado_en'] != null ? DateTime.parse(json['actualizado_en']) : null,
      actualizadoPor: json['actualizado_por'],
    );
  }

  Map<String, dynamic> toJson() {
    final data = <String, dynamic>{
      'codigo': codigo,
      'nombre': nombre,
      'descripcion': descripcion,
      'activo': activo,
    };
    if (idZona != null) data['id_zona'] = idZona;
    if (creadoPor != null) data['creado_por'] = creadoPor;
    if (actualizadoPor != null) data['actualizado_por'] = actualizadoPor;
    return data;
  }
}
